src/deepsparse/transformers/metrics.py

```diff
@@ -122,22 +122,6 @@ def add_batch(self, predictions: List[str]):
 
             logits = out.logits
 
-            if not self._pipeline.has_cache:
```

Comment:
why are we moving this logic from the metric to the evaluation script? this does not make sense to me from the design point of view and would only add unnecessary code.

Reply:
I tried to follow the ppl and metrics design here, but got bad results. I did it this way to match how it was done from the pytorch flow, and mainly to make sure we got the right ppl numbers; plus for the sake of time. If you have concrete suggestions how to move these into the framework, please help.

Reply:
I think this pathway is important to support perplexity evaluation of non-cache models. Wouldn't we include likelihood for pad tokens if we remove this? For testing, did you test non-cache models?